pkg/controllers/node/initialization.go

```diff
@@ -73,32 +73,41 @@ func (r *Initialization) getInstanceType(ctx context.Context, provisioner *v1alp
 // This method handles both nil provisioners and nodes without extended resources gracefully.
 func (r *Initialization) isInitialized(n *v1.Node, provisioner *v1alpha5.Provisioner, instanceType cloudprovider.InstanceType) bool {
 	// fast checks first
-	return node.GetCondition(n, v1.NodeReady).Status == v1.ConditionTrue &&
-		isStartupTaintRemoved(n, provisioner) && isExtendedResourceRegistered(n, instanceType)
+	if node.GetCondition(n, v1.NodeReady).Status != v1.ConditionTrue {
+		return false
+	}
+	if _, ok := IsStartupTaintRemoved(n, provisioner); !ok {
+		return false
+	}
+
+	if _, ok := IsExtendedResourceRegistered(n, instanceType); !ok {
+		return false
+	}
+	return true
 }
 
-// isStartupTaintRemoved returns true if there are no startup taints registered for the provisioner, or if all startup
+// IsStartupTaintRemoved returns true if there are no startup taints registered for the provisioner, or if all startup
 // taints have been removed from the node
-func isStartupTaintRemoved(node *v1.Node, provisioner *v1alpha5.Provisioner) bool {
+func IsStartupTaintRemoved(node *v1.Node, provisioner *v1alpha5.Provisioner) (*v1.Taint, bool) {
 	if provisioner != nil {
 		for _, startupTaint := range provisioner.Spec.StartupTaints {
 			for i := 0; i < len(node.Spec.Taints); i++ {
 				// if the node still has a startup taint applied, it's not ready
 				if startupTaint.MatchTaint(&node.Spec.Taints[i]) {
-					return false
+					return &node.Spec.Taints[i], false
 				}
 			}
 		}
 	}
-	return true
+	return nil, true
 }
 
-// isExtendedResourceRegistered returns true if there are no extended resources on the node, or they have all been
+// IsExtendedResourceRegistered returns true if there are no extended resources on the node, or they have all been
 // registered by device plugins
-func isExtendedResourceRegistered(node *v1.Node, instanceType cloudprovider.InstanceType) bool {
+func IsExtendedResourceRegistered(node *v1.Node, instanceType cloudprovider.InstanceType) (v1.ResourceName, bool) {
```

Comment:
Return back all extended resources that aren't registered?

Reply:
The other caller of this method doesn't care about the names of any, only if one exists so I just had this return the first.  This error only occurs infrequently due to a configuration and it's unlikely there are multiple resources that don't register.  If there are, one the first one is fixed this will report the next.